flang/test/Lower/intrinsic-procedures.f90

```diff
@@ -783,6 +783,49 @@ subroutine product_test2(a,r)
 ! CHECK: fir.call @_FortranACppProductComplex10(%[[a4]], %[[a5]], %{{.*}}, %{{.*}}, %[[a7]], %8) : (!fir.ref<complex<f80>>, !fir.box<none>, !fir.ref<i8>, i32, i32, !fir.box<none>) -> ()
 end
 
+! CHECK-LABEL: func @_QPrandom_test
+subroutine random_test
+  integer ss, vv(40)
+  real rr, aa(5)
+  ! CHECK: fir.call @_FortranARandomInit(%true{{.*}}, %false{{.*}}) : (i1, i1) -> none
+  call random_init(.true., .false.)
+  ! CHECK: fir.call @_FortranARandomSeedSize(%{{[0-9]+}}, %{{[0-9]+}}, %c{{.*}}) : (!fir.box<none>, !fir.ref<i8>, i32) -> none
+  call random_seed(size=ss)
+  ! CHECK: fir.call @_FortranAioBeginExternalListOutput
+  ! CHECK: fir.call @_FortranAioOutputAscii
+  ! CHECK: fir.call @_FortranAioOutputInteger64
+  ! CHECK: fir.call @_FortranAioEndIoStatement
```

Comment:
I would not check the IO statement here, this is not relevant to the intrinsic being tested here.